mobile-app/app/screens/AppNavigator/screens/Portfolio/components/ProposalURLInput.tsx

```diff
@@ -0,0 +1,151 @@
+import { translate } from "@translations";
+import { tailwind } from "@tailwind";
+import { ThemedIcon, ThemedTextV2 } from "@components/themed";
+import { WalletTextInputV2 } from "@components/WalletTextInputV2";
+import { useEffect, useState } from "react";
+import { View } from "react-native";
+import { Controller, useForm } from "react-hook-form";
+
+interface ProposalURLInputProps {
+  urlValidity: (isValid: boolean) => void;
+  onChangeUrlInput: (text: string) => void;
+}
+
+export function ProposalURLInput({
+  urlValidity,
+  onChangeUrlInput,
+}: ProposalURLInputProps): JSX.Element {
+  const [status, setStatus] = useState<ProposalInputStatus>(
+    ProposalInputStatus.Default
+  );
+
+  // form
+  const {
+    control,
+    setValue,
+    getValues,
+    trigger,
+    formState: { isValid },
+  } = useForm({
+    mode: "onChange",
+  });
+  const url = getValues("url");
+  const hasInput = url !== "" && url !== undefined;
+
+  useEffect(() => {
+    async function checkUrlValidity() {
```

Comment:
do we have any promises in this function?

Reply:
nope

Reply:
then do we need `async`?

Reply:
ah good catch, removed `async`, thanks!